src/Simplify.cpp

```diff
@@ -31,5923 +13,331 @@ using std::pair;
-            // In ternary expressions, pull constants outside
-            if (is_simple_const(b)) {
-                return mutate(add_a->a + (add_a->b - b));
-            } else {
-                return mutate((add_a->a - b) + add_a->b);
-            }
-        } else if (sub_a &&
-                   sub_b &&
-                   is_const(sub_a->a) &&
-                   is_const(sub_b->a)) {
-            // (c1 - a) - (c2 - b) -> (b - a) + (c1 - c2)
-            return mutate((sub_b->b - sub_a->b) + (sub_a->a - sub_b->a));
-        } else if (sub_b) {
-            // a - (b - c) -> a + (c - b)
-            return mutate(a + (sub_b->b - sub_b->a));
-        } else if (mul_b &&
-                   is_negative_negatable_const(mul_b->b)) {
-            // a - b*-x -> a + b*x
-            return mutate(a + mul_b->a * (-mul_b->b));
-        } else if (mul_b &&
-                   !is_const(a) &&
-                   equal(a, mul_b->a) &&
-                   no_overflow(op->type)) {
-            // a - a*b -> a*(1 - b)
-            return mutate(a * (make_one(op->type) - mul_b->b));
-        } else if (mul_b &&
-                   !is_const(a) &&
-                   equal(a, mul_b->b) &&
-                   no_overflow(op->type)) {
-            // a - b*a -> (1 - b)*a
-            return mutate((make_one(op->type) - mul_b->a) * a);
-        } else if (mul_a &&
-                   !is_const(b) &&
-                   equal(mul_a->a, b) &&
-                   no_overflow(op->type)) {
-            // a*b - a -> a*(b - 1)
-            return mutate(mul_a->a * (mul_a->b - make_one(op->type)));
-        } else if (mul_a &&
-                   !is_const(b) &&
-                   equal(mul_a->b, b) &&
-                   no_overflow(op->type)) {
-            // a*b - b -> (a - 1)*b
-            return mutate((mul_a->a - make_one(op->type)) * b);
-        } else if (add_b &&
-                   is_simple_const(add_b->b)) {
-            return mutate((a - add_b->a) - add_b->b);
-        } else if (sub_a &&
-                   is_simple_const(sub_a->a) &&
-                   is_simple_const(b)) {
-            return mutate((sub_a->a - b) - sub_a->b);
-        } else if (mul_a &&
-                   mul_b &&
-                   equal(mul_a->a, mul_b->a)) {
-            // Pull out common factors a*x + b*x
-            return mutate(mul_a->a * (mul_a->b - mul_b->b));
-        } else if (mul_a &&
-                   mul_b &&
-                   equal(mul_a->b, mul_b->a)) {
-            return mutate(mul_a->b * (mul_a->a - mul_b->b));
-        } else if (mul_a &&
-                   mul_b &&
-                   equal(mul_a->b, mul_b->b)) {
-            return mutate(mul_a->b * (mul_a->a - mul_b->a));
-        } else if (mul_a &&
-                   mul_b &&
-                   equal(mul_a->a, mul_b->b)) {
-            return mutate(mul_a->a * (mul_a->b - mul_b->a));
-        } else if (add_a &&
-                   add_b &&
-             
```

Comment:
This will need to change to debug(1) before landing, otherwise we'll be getting lots of confused bug reports from end-users